service/history/replication/bi_direction_stream.go

```diff
@@ -192,9 +197,20 @@ func (s *BiDirectionStreamImpl[Req, Resp]) recvLoop() {
 			var errResp Resp
 			s.channel <- StreamResp[Resp]{
 				Resp: errResp,
-				Err:  err,
+				Err:  NewStreamError("BiDirectionStream recv error", err),
 			}
 			return
 		}
 	}
 }
+
+func (e *StreamError) Error() string {
+	return fmt.Sprintf("StreamError: %s | GRPC Error: %v", e.Message, e.err)
```

Comment:
is `err` always going to be GRPC error? change `e.err` to `e.cause` ?

Reply:
This file is the wrap of gRPC stream. So all error in this file is not the application logic error, so they can be all considered as stream error.

Yeah, `e.cause` make sense.